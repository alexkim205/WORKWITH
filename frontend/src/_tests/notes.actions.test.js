import createMockStore from '../_config/mockStore.config';
import {
  getNotesByProject,
  getNotes,
  getNote,
  createNote,
  updateNote,
  deleteNote
} from '../_actions/notes.actions';
import testConstants from '../_constants/test.constants';

const {
  notes: { note, newNote, newUpdateNote },
  projects: { project }
} = testConstants;

describe('Note Actions', () => {
  let store;
  beforeAll(() => {
    store = createMockStore();
  });
  afterEach(() => {
    store.clearActions();
  });
  it('it should GET notes by project', async () => {
    await store.dispatch(getNotesByProject(project._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should GET notes', async () => {
    await store.dispatch(getNotes());
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should GET a note', async () => {
    await store.dispatch(getNote(note._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should CREATE a note', async () => {
    await store.dispatch(createNote(newNote));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should UPDATE a note', async () => {
    await store.dispatch(updateNote(note._id, newUpdateNote));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should DELETE a note', async () => {
    await store.dispatch(deleteNote(note._id));
    expect(store.getActions()).toMatchSnapshot();
  });
});
