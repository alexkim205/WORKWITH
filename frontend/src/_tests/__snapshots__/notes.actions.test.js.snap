// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Actions it should CREATE a note 1`] = `
Array [
  Object {
    "type": "notes/create-note-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 1,
      "_id": "5e02f1b53e741a3e18c4027b",
      "authors": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "body": "I like my new body.",
      "createdAt": "2019-12-25T05:20:53.800Z",
      "deleted": false,
      "minimized": true,
      "private": false,
      "projectId": "5e02efdd94748b21482e6010",
      "taggedUsers": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "title": "New Title",
      "updatedAt": "2019-12-30T09:07:10.417Z",
    },
    "type": "notes/create-note-success",
  },
]
`;

exports[`Note Actions it should DELETE a note 1`] = `
Array [
  Object {
    "type": "notes/delete-note-pending",
  },
  Object {
    "error": null,
    "payload": undefined,
    "type": "notes/delete-note-success",
  },
]
`;

exports[`Note Actions it should GET a note 1`] = `
Array [
  Object {
    "type": "notes/get-note-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 1,
      "_id": "5e02f1b53e741a3e18c4027b",
      "authors": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "body": "I like my new body.",
      "createdAt": "2019-12-25T05:20:53.800Z",
      "deleted": false,
      "minimized": true,
      "private": false,
      "projectId": "5e02efdd94748b21482e6010",
      "taggedUsers": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "title": "New Title",
      "updatedAt": "2019-12-30T09:07:10.417Z",
    },
    "type": "notes/get-note-success",
  },
]
`;

exports[`Note Actions it should GET notes 1`] = `
Array [
  Object {
    "type": "notes/get-notes-pending",
  },
  Object {
    "error": null,
    "payload": Array [
      Object {
        "__v": 1,
        "_id": "5e02f1b53e741a3e18c4027b",
        "authors": Array [
          "5e02eeaf6ebf6452d426c135",
        ],
        "body": "I like my new body.",
        "createdAt": "2019-12-25T05:20:53.800Z",
        "deleted": false,
        "minimized": true,
        "private": false,
        "projectId": "5e02efdd94748b21482e6010",
        "taggedUsers": Array [
          "5e02eeaf6ebf6452d426c135",
        ],
        "title": "New Title",
        "updatedAt": "2019-12-30T09:07:10.417Z",
      },
    ],
    "type": "notes/get-notes-success",
  },
]
`;

exports[`Note Actions it should GET notes by project 1`] = `
Array [
  Object {
    "type": "notes/get-project-notes-pending",
  },
  Object {
    "error": null,
    "payload": Array [
      Object {
        "__v": 1,
        "_id": "5e02f1b53e741a3e18c4027b",
        "authors": Array [
          "5e02eeaf6ebf6452d426c135",
        ],
        "body": "I like my new body.",
        "createdAt": "2019-12-25T05:20:53.800Z",
        "deleted": false,
        "minimized": true,
        "private": false,
        "projectId": "5e02efdd94748b21482e6010",
        "taggedUsers": Array [
          "5e02eeaf6ebf6452d426c135",
        ],
        "title": "New Title",
        "updatedAt": "2019-12-30T09:07:10.417Z",
      },
    ],
    "type": "notes/get-project-notes-success",
  },
]
`;

exports[`Note Actions it should UPDATE a note 1`] = `
Array [
  Object {
    "type": "notes/update-note-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 1,
      "_id": "5e02f1b53e741a3e18c4027b",
      "authors": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "body": "I like my new body.",
      "createdAt": "2019-12-25T05:20:53.800Z",
      "deleted": false,
      "minimized": true,
      "private": false,
      "projectId": "5e02efdd94748b21482e6010",
      "taggedUsers": Array [
        "5e02eeaf6ebf6452d426c135",
      ],
      "title": "New Title",
      "updatedAt": "2019-12-30T09:07:10.417Z",
    },
    "type": "notes/update-note-success",
  },
]
`;
